### PR TITLE
feat: Admin Panel Route Guard

### PR DESCRIPTION
**Pull request by bsilkyn**
A route guard for the admin panel would be quite handy to prevent non-admins from reaching it in the first place

### CODE REVIEW

**Comment by bsilkyn** [comment]
Wait with merging until demo is done

**Comment by bsilkyn** [comment]
It can be merged now, after being reviewed

File: frontend/src/router/router.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import VerifyView from '@/views/authentication/VerifyView.vue';
 import { type RouteRecordRaw, createWebHistory, createRouter } from 'vue-router';
 import { AuthenticationGuard } from '@/router/guards/authentication.guard.ts';
 import { LogoutGuard } from '@/router/guards/logout.guard.ts';
+import { AdminGuard } from '@/router/guards/admin.guard.ts';
```
Comment (tyboro2002) [inline frontend/src/router/router.ts:15]: from where is this file this is not in the repo

**Review by tyboro2002 — CHANGES_REQUESTED** [review]

**Comment by tyboro2002** [comment]
> It can be merged now, after being reviewed

you can also set the pr to a normal pr instead of a draft to notify this.

**Review by francisvaut — APPROVED** [review]
This is better.

**Pull request merged**